src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -610,6 +621,14 @@ GenerateSizeQueryOnMultiplePlacements(List *shardIntervalList,
 	ShardInterval *shardInterval = NULL;
 	foreach_ptr(shardInterval, shardIntervalList)
 	{
+		if (optimizePartitionCalculations && PartitionTable(shardInterval->relationId))
```

Comment:
Interesting, we also don't seem to have coverage of this case here

Reply:
It's actually pretty expected. Because we don't send the child tables to this function. They are all eliminated in the new function `ColocatedNonPartitionShardIntervalList`

Reply:
The actual problem is, it seems impossible to cover here :)